File: C++ code/lists/maximum_ascending_subarray_sum.cpp
class Solution {
public:
    int maxAscendingSum(vector<int>& nums) {
        int curr = nums[0];
        int max_res = curr;

        for(int i=1; i<size(nums); i++){
            if(nums[i-1] < nums[i]){
                curr += nums[i];
            }
            else{
                curr = nums[i];
            }  
            max_res = std::max(max_res, curr);
        }
        return max_res; 
    }
};

File: lists/maximum_ascending_subarray_sum.py
class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:
        curr = nums[0]
        max_res = curr

        for i in range(1, len(nums)):
            if nums[i-1] < nums[i]:
                curr += nums[i]
            else:
                curr = nums[i]
            max_res = max(curr, max_res)
        return max_res

        
        